Services/Implementations/TaskService.cs

```diff
@@ -0,0 +1,53 @@
+namespace Services.Implementations;
+using Repositories;
+using DataTransfer.Objects;
+
+public class TaskService : ITaskService
+{
+	private readonly ITaskRepository taskRepo;
+
+	public TaskService(ITaskRepository taskRepo)
+	{
+		this.taskRepo = taskRepo;
+	}
+
+	public IEnumerable<TaskDto> GetTasks()
+	{
+		return taskRepo.GetTasks();
+	}
+
+    public IEnumerable<TaskDto> GetTasksByDeadline()
+	{
+		throw new NotImplementedException();
+	}
+
+    public IEnumerable<TaskDto> GetOverdueTasks()
+	{
+		return taskRepo.GetTasks().Where(
+			task => task.Deadline > DateTime.Now
```

Comment:
It is better to move DateTime.Now to local variable, as sometimes evaluation of properties may be impossible when  converting expression tree to request.